src/awkward/contents/unionarray.py

```diff
@@ -1586,7 +1586,7 @@ def _remove_structure(
         # backends with concrete data
         for i in range(self._tags.length):
             content = (
-                self._contents[self._tags.data[i]]
+                self._contents[int(self._tags.data[i])]
```

Comment:
Looks good to me, I was just curious if it should maybe be `self._tags.data[i].item()` instead (or is this the same?)?

Reply:
Thanks, yes, I think, it's the same result. I'll check what we do in other places.

Reply:
I saw e.g.: https://github.com/scikit-hep/awkward/blob/58eea9bc541e0f672468a846f80c0c980a90aa8d/src/awkward/_nplikes/cupy.py#L75 which uses `.item()` and thought that may be the preferred way for CuPy?

Reply:
Oh, wait, actually I'm confused, shouldn't this do that already automatically: https://github.com/scikit-hep/awkward/blob/58eea9bc541e0f672468a846f80c0c980a90aa8d/src/awkward/index.py#L236 ?

Reply:
Could you try `self._tags[i]` instead of `int(self._tags.data[i])` ?

Reply:
Thanks! Done.